plugins/modules/ntnx_vms_clone.py

```diff
@@ -0,0 +1,325 @@
+                    ],
+                    "gpu_list": [],
+                    "hardware_clock_timezone": "UTC",
+                    "host_reference": {
+                        "kind": "host",
+                        "name": "10.46.136.28",
+                        "uuid": "c91be067-9885-4cbb-92cc-876242c0c396"
+                    },
+                    "hypervisor_type": "AHV",
+                    "is_agent_vm": false,
+                    "machine_type": "PC",
+                    "memory_size_mib": 4096,
+                    "nic_list": [],
+                    "num_sockets": 1,
+                    "num_threads_per_core": 1,
+                    "num_vcpus_per_socket": 1,
+                    "power_state": "ON",
+                    "power_state_mechanism": {
+                        "guest_transition_config": {
+                            "enable_script_exec": false,
+                            "should_fail_on_script_failure": false
+                        },
+                        "mechanism": "HARD"
+                    },
+                    "protection_type": "UNPROTECTED",
+                    "serial_port_list": [],
+                    "vga_console_enabled": true,
+                    "vnuma_config": {
+                        "num_vnuma_nodes": 0
+                    }
+                },
+                "state": "COMPLETE"
+            }
+vm_uuid:
+  description: The cloned vm uuid
+  returned: always
+  type: str
+  sample: "2b011db0-4d44-43e3-828a-d0a32dab340c"
+task_uuid:
+  description: The task uuid for the clone
+  returned: always
+  type: str
+  sample: "82c5c1d3-eb6a-406a-8f58-306028099d21"
+"""
+
+from ..module_utils import utils  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.prism.vm_base_module import VMBaseModule  # noqa: E402
+from ..module_utils.prism.vms import VM  # noqa: E402
+
+
+def clone_vm(module, result):
+    vm_uuid = module.params["vm_uuid"]
+
+    vm = VM(module)
+
+    spec, error = vm.get_clone_spec()
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating VM Spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+
+    resp = vm.clone(spec)
+
+    result["changed"] = True
+    result["response"] = resp
+    result["task_uuid"] = resp["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+        resp = vm.read(vm_uuid)
+        result["response"] = resp
+
+
+def wait_for_task_completion(module, result, raise_error=True):
+    task = Task(module)
+    task_uuid = result["task_uuid"]
+    resp = task.wait_for_completion(task_uuid, raise_error=raise_error)
+    result["response"] = resp
+    if not result.get("vm_uuid") and resp.get("entity_reference_list"):
+        result["vm_uuid"] = resp["entity_reference_list"][0]["uuid"]
+
+
+def run_module():
+    module = VMBaseModule(supports_check_mode=True)
```

Comment:
Confirm Once. We can override name and other params while cloning. Pleas check https://www.nutanix.dev/api_references/prism-central-v3/#/b3A6MjU1ODc4NDA-clone-a-vm

Reply:
it's supporting